lib/Resque/Job.php

```diff
@@ -242,6 +242,11 @@ public function fail($exception)
 				$this->queue
 			);
 		}
+		
+		if(!empty($job->payload['id'])) {
```

Comment:
There is a lot to be said about this but primarily there is no reference to $job at this point.
So it should at least be $this.
My other comment would be that cleaning up redis Id's happens in the worker and failing a job also happens in the worker. So to me it makes more sense to cleanup this id in the worker.

Reply:
I've fixed the variable name.

I could move this code, and the stat incrementations to a new location that listens for the onFailure event?